src/api/typed-api.ts

```diff
@@ -0,0 +1,102 @@
+/** Type-safe access to the RefStudio HTTP APIs. */
```

Comment:
It would be amazing to have a blog post about this code and how typescript can help in such use-cases.

Reply:
Or a book chapter? :)